cql3/Cql.g

```diff
@@ -389,6 +389,7 @@ selectStatement returns [std::unique_ptr<raw::select_statement> expr]
     }
     : K_SELECT (
                 ( K_JSON { statement_subtype = raw::select_statement::parameters::statement_subtype::JSON; } )?
+                ( K_MUTATION K_FRAGMENTS { statement_subtype = raw::select_statement::parameters::statement_subtype::MUTATION_FRAGMENTS; } )?
                 ( K_DISTINCT { is_distinct = true; } )?
```

Comment:
Another idea: mutilate the FROM clause.

```cql
SELECT *
FROM mutation_fragments(tab)
WHERE ...
```

Reply:
This could also work. But note that I also want a working `WHERE` clause. One of the big reasons why I stuck with CQL for this feature instead of API, is the built-in data manipulation capabilities in CQL. I expect that in the vast majority of cases, we will not want to read everything from this, but restrict it in different ways. E.g. read partition x from memtable, read all partitions from cache, read only tombstones from cache, etc.

Reply:
Ah, I missed that you talked about the `FROM` clause, not the `WHERE` clause.
Let me try to see how a "mutilated" `FROM` this would look like. I like that it emphasizes that we are doing something to the entire table.

Reply:
This was quite easy:
```cql
cqlsh> select * from mutation_fragments(ks.tbl) where pk = 0;

 pk | mutation_source_kind | partition_region | ck | position_weight | mutation_fragment_kind | value
----+----------------------+------------------+----+-----------------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  0 |              sstable |                0 |    |            null |        partition start |                                                                                                                                                                                                                                           null
  0 |              sstable |                2 |  0 |               0 |         clustering row | {mutation_fragment: clustering row {position: clustered, ckp{000400000000}, 0} {clustering_row: ck 000400000000 dr {deletable_row: {row_marker: 1687357831647535 0 0} {{row:\n    v{ atomic_cell{0,ts=1687357831647535,expiry=-1,ttl=0} }}}}}}
  0 |              sstable |                2 |  1 |               0 |         clustering row | {mutation_fragment: clustering row {position: clustered, ckp{000400000001}, 0} {clustering_row: ck 000400000001 dr {deletable_row: {row_marker: 1687357833978414 0 0} {{row:\n    v{ atomic_cell{0,ts=1687357833978414,expiry=-1,ttl=0} }}}}}}
  0 |              sstable |                2 |  2 |               0 |         clustering row | {mutation_fragment: clustering row {position: clustered, ckp{000400000002}, 0} {clustering_row: ck 000400000002 dr {deletable_row: {row_marker: 1687357835597943 0 0} {{row:\n    v{ atomic_cell{0,ts=1687357835597943,expiry=-1,ttl=0} }}}}}}
  0 |              sstable |                3 |    |            null |          partition end |                                                                                                                                                                                                                                           null

(5 rows)
```
What do you think?

Reply:
Here is the diff I had to do (compared to the current version):
```diff
diff --git a/cql3/Cql.g b/cql3/Cql.g
index fcd8d75ce4..1a2eb2205e 100644
--- a/cql3/Cql.g
+++ b/cql3/Cql.g
@@ -389,11 +389,13 @@ selectStatement returns [std::unique_ptr<raw::select_statement> expr]
     }
     : K_SELECT (
                 ( K_JSON { statement_subtype = raw::select_statement::parameters::statement_subtype::JSON; } )?
-                ( K_MUTATION K_FRAGMENTS { statement_subtype = raw::select_statement::parameters::statement_subtype::MUTATION_FRAGMENTS; } )?
                 ( K_DISTINCT { is_distinct = true; } )?
                 sclause=selectClause
                )
-      K_FROM cf=columnFamilyName
+      K_FROM (
+                cf=columnFamilyName
+                | K_MUTATION_FRAGMENTS '(' cf=columnFamilyName ')' { statement_subtype = raw::select_statement::parameters::statement_subtype::MUTATION_FRAGMENTS; }
+             )
       ( K_WHERE w=whereClause { wclause = std::move(w); } )?
       ( K_GROUP K_BY gbcolumns=listOfIdentifiers)?
       ( K_ORDER K_BY orderByClause[orderings] ( ',' orderByClause[orderings] )* )?
@@ -2069,8 +2071,7 @@ basic_unreserved_keyword returns [sstring str]
         | K_DESCRIBE
         | K_DESC
         | K_EXECUTE
-        | K_MUTATION
-        | K_FRAGMENTS
+        | K_MUTATION_FRAGMENTS
         ) { $str = $k.text; }
     ;
 
@@ -2279,8 +2280,7 @@ K_PRUNE:       P R U N E;
 
 K_EXECUTE:     E X E C U T E;
 
-K_MUTATION:    M U T A T I O N;
-K_FRAGMENTS:   F R A G M E N T S;
+K_MUTATION_FRAGMENTS:    M U T A T I O N '_' F R A G M E N T S;
 
 // Case-insensitive alpha characters
 fragment A: ('a'|'A');
```

Reply:
I think I like this version better, and there is less chance of mixing it up with columns, as you pointed out.

Reply:
Yes

Reply:
FWIW, `SELECT mutation_fragments(*) FROM tab` makes even more sense to me, since returning mutation_fragments instead of the actual data is a function/transformation of the returned data.
With that you can also restrict the columns you want mutation fragments for, e.g. `SELECT mutation_fragments(my, columns) FROM tab WHERE my_row = 'this_row' LIMIT 10`.

Reply:
Currently the output schema of this statement is not that of the underlying table. So you cannot put columns of the underlying table in the selector, you can only use columns of the output schema, so what you propose above doesn't work.

Reply:
Yes, a straightforward reading of SELECT mutation_fragments(*) means "transform all columns of the table into one column", but that's not what we want.

The new syntax means "transform the table into some view, then select all columns from the view.

In that sense, SELECT MUTATION FRAGMENTS * FROM tab would also work, since "MUTATION FRAGMENTS" modifies the SELECT, not the *.